lib/rts2tel/teld.cpp

```diff
@@ -2377,7 +2381,20 @@ int Telescope::startResyncMove (rts2core::Connection * conn, int correction)
 		}
 	}
 
-	// all checks done, give proper info and set TEL_ state and BOP
```

Comment:
I suggest keeping the original state of this (rejecting this particular change, ie following 14 lines).
I understand this was ment for speeding up the movement initiation, but, the original code is not time-consuming and sets all the complex teld states before the real movement is really initiated (which COULD be time-consuming).
Also, you didn't delete the original part of the code (startResync () etc below), which is probably just a consequence of incorrect merging.

Reply:
Yes, the original commit is here - https://github.com/karpov-sv/rts2/commit/db0690b385668428f304f831399f33b43b757bad
It moves startResync() piece, and explains why it was done - not to speed things up, but to prevent accidental sending of EVENT_MOVE_OK before EVENT_MOVE_FAILED in case of movement error.

Reply:
Well, this definitely should not happen.
The EVENT_MOVE_* are part of executor code, there is a need to revise the states of the teld at all (to set the error states for more cases than it's now, to be consistent with the executor). I know about this issue - and I'll look at it soon.
I doubt that just changing the order (as it is in this commit) will solve all the issues.
For now, I suggest keeping this specific part of the code at the state of the main branch.
Or at least, postpone the merge a bit, I'll try to examine it ASAP :-).